cfpb/consumerfinance.gov
# Enable FilterableList ordering from Wagtail Admin

We're noticing that in some cases we prefer content be ordered by publication date, and others to be ordered by relevance. This will allow content managers on the admin side to determine which ordering makes the most sense for their pages.


---


## Additions

- order_by choice block to allow selecting which method for ordering your results.


## Removals

-


## Changes

- Will always order by date published unless we submit a search term and have opted for relevance ordering.


## How to test this PR

1. Visit a filterable page such as the Blog Page.
2.  Ensure initial content is ordered by date
3.  Search any term such as "mortgage"
4.  Results should be returned by publication date
5.  Update the order_by selection in the admin panel
6.  Visit the blog landing page and confirm results are still ordered by date
7.  Filter by a category or topic and ensure results are still ordered by date
8.  Search a keyword such as mortgage and confirm results come back by relevance (or confirm they are not returned in date order is likely easier).


## Screenshots
![image](https://user-images.githubusercontent.com/16838497/111687159-560cf780-8800-11eb-9b2a-21f31c9eafd7.png)


## Notes and todos

-


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
